src/methods/sign.ts

```diff
@@ -3,26 +3,41 @@ import { Keyring } from "@polkadot/keyring";
 import type { KeyringPair } from "@polkadot/keyring/types";
 import { cryptoWaitReady } from "@polkadot/util-crypto";
 import prompts from "prompts";
+import { NetworkType } from "./types";
 
 // TODO display payload content
-export async function sign(type: string, privKey: string) {
+export async function sign(
+  type: NetworkType,
+  privKey: string,
+  prompt: boolean,
+  message?: string
+): Promise<string> {
   if (!["ethereum", "sr25519"].includes(type)) {
     throw new Error("Type is not supported");
   }
   await cryptoWaitReady();
   let keyring: Keyring = new Keyring({ type: type === "ethereum" ? "ethereum" : "sr25519" });
   const signer: KeyringPair = keyring.addFromSeed(hexToU8a(privKey));
-  const response = await prompts({
-    type: "text",
-    name: "message",
-    message: "Please enter payload",
-    validate: (value) => true, //value < 18 ? `Nightclub is 18+ only` : true
-  });
+  let msg: string = "";
```

Comment:
General tips:
- when the type is basic and given in the declaration, there is no real need to add it.  Here
`let msg = "";` is enough for readability

- It is better to keep all the variable `const`, it makes it easier to read it. Ex:
```
if (!prompt && !message) {
  throw new Error("sign must either provide message or use prompt");
}

const msg = message ||
  (await prompts({
      type: "text",
      name: "message",
      message: "Please enter payload",
      validate: (value) => true,
    })).message;
```
(I don't think "//value < 18 ? `Nightclub is 18+ only` : true" is a necessary comment also :) )

Reply:
To clarify, the difficulty with `let` is that you don't know in the code where the value is modified, making it harder to read.

Reply:
Thank you for the advice :)